security/pkg/k8s/chiron/utils.go

```diff
@@ -252,12 +254,14 @@ func submitCSR(clientset clientset.Interface,
 		v1beta1req, err := clientset.CertificatesV1beta1().CertificateSigningRequests().Create(context.TODO(), v1beta1csr, metav1.CreateOptions{})
 		if err == nil {
 			return csrName, nil, v1beta1req, nil
-		} else if apierrors.IsAlreadyExists(err) {
+		}
+		lastErr = err
+		if apierrors.IsAlreadyExists(err) {
 			csrName = ""
 		}
 	}
-	log.Errorf("retry attempts exceeded when creating csr request %v: %v", csrName, err)
-	return "", nil, nil, err
+	log.Errorf("retry attempts exceeded when creating csr request %v", csrName)
```

Comment:
The original log statement outputs err in Errorf. Why does the new log statement not output the lastErr in Errof?

Reply:
It will eventually be logged at https://github.com/istio/istio/blob/23c56d13515f8c415bc914082a88e4c296d4efde/pilot/cmd/pilot-discovery/app/cmd.go#L94, e.g. this the log lines if I modify the code in this function to return a dummy error:

```
2021-08-30T22:37:03.612350Z     info    initializing Istiod DNS certificates host: istiod.istio-system.svc, custom host: 
2021-08-30T22:37:03.612410Z     info    Generating K8S-signed cert for [istiod.istio-system.svc istiod-remote.istio-system.svc istio-pilot.istio-system.svc]
2021-08-30T22:37:03.906321Z     error   failed to create discovery service: failed generating key and cert by kubernetes: unable to submit CSR request (). Error: dummy error
Error: failed to create discovery service: failed generating key and cert by kubernetes: unable to submit CSR request (). Error: dummy error
```

Reply:
Thanks! Approved.